editor/editor_export.cpp

```diff
@@ -1776,19 +1776,46 @@ void EditorExportTextSceneToBinaryPlugin::_export_file(const String &p_path, con
 	if (!convert) {
 		return;
 	}
-	String tmp_path = EditorSettings::get_singleton()->get_cache_dir().plus_file("tmpfile.res");
-	Error err = ResourceFormatLoaderText::convert_file_to_binary(p_path, tmp_path);
```

Comment:
If this is desired, I think compression should be added to this `ResourceFormatLoaderText::convert_file_to_binary` method.

With your change, every converted scene has to be loaded and saved but the commit message for 251433847f85124d87f193ded4cd353e1f5bf5f5, which added `ResourceFormatLoaderText::convert_file_to_binary`, mentions that "This method works by directly converting text to binary, so the scene does not need to be loaded and saved". There was probably a reason by @reduz to do it this way (I guess it's faster?)

Reply:
"This method works by directly converting text to binary, so the scene does not need to be loaded and saved" - internally it is loaded and after conversion saved as binary `tmpfile.res` and then we are loading it with `FileAccess::get_file_as_array()`. I'm not sure if it's a good approach to duplicate functionality instead of reusing `ResourceSaver::save()` (which is used by editor when we save the scene or other resources).